### PR TITLE
Restore credit_flow between AMQP 0.9.1 channel/MQTT connection -> CQ processes

### PR DESCRIPTION
**Pull request by gomoripeti**
## Proposed Changes

The credit_flow between publishing AMQP 0.9.1 channel (or MQTT
connection) and (non-mirrored) classic queue processes was
unintentionally removed in 4.0 together with anything else related to
CQ mirroring.

By default we restore the 3.x behaviour for non-mirored classic
queues. It is possible to disable flow-control (the earlier 4.0.x
behaviour) with the new env `classic_queue_flow_control`. In 3.x this
was possible with the config `mirroring_flow_control`.

Most of the changes are reverted portions of 3bbda5bdba799f1d4ec3c2ab055b4177f35ae78c, except the new function `rabbit_misc:process_info/2` and the new test cases `classic_queue_SUITE:classic_queue_flow_control_(enabled|disabled)`

Fixes #12885

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

File: deps/rabbit_common/src/rabbit_misc.erl [hunk 1]
```diff
@@ -812,6 +813,23 @@ is_process_alive(Pid) ->
...
+      Result :: {atom() | tuple(), term()} | [{atom() | tuple(), term()}].
+process_info(Pid, Items) when node(Pid) =:= node() ->
+    erlang:process_info(Pid, Items);
+process_info(Pid, Items) ->
+    Node = node(Pid),
+    case lists:member(Node, [node() | nodes(connected)]) of
+        true ->
+            rpc:call(Node, erlang, process_info, [Pid, Items]);
+        _ ->
+            {badrcp, nodedown}
```
Comment (lukebakken) [inline deps/rabbit_common/src/rabbit_misc.erl:830]: `badrcp` -> `badrpc`

Comment (lukebakken) [inline deps/rabbit_common/src/rabbit_misc.erl:830]:
```suggestion
            {badrpc, nodedown}
```

**Review by michaelklishin — CHANGES_REQUESTED** [review]
`web_mqtt_shared_SUITE > mqtt > cluster_size_3 > v4 > flow_classic_queue` fails with an `undef` repeatedly, for example in https://github.com/rabbitmq/rabbitmq-server/actions/runs/12248903112/job/34169379099?pr=12907#step:10:5274

**Review by michaelklishin — COMMENTED** [review]
The CI failure was a matter of propagating one more test to the `mqtt_shared_SUITE` https://github.com/rabbitmq/rabbitmq-server/pull/12907/commits/f3540ee7d29dc221f2c36c287001b06af8218ddd.

**Comment by michaelklishin** [comment]
https://github.com/rabbitmq/rabbitmq-server/pull/12907 is green so we can proceed to other stages of reviews and testing the workload from https://github.com/rabbitmq/rabbitmq-server/discussions/12885 against this branch.

**Comment by gomoripeti** [comment]
I did not realise why the shared_SUITE is called shared. I thought it is shared between MQTT v3 and v5.
Thank you, Michael, for fixing it!

**Review by lhoguin — APPROVED** [review]

**Comment by michaelklishin** [comment]
Merged in https://github.com/rabbitmq/rabbitmq-server/pull/12907. Thank you, @gomoripeti.

**Pull request closed without merging**